pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -304,6 +327,80 @@ async def prep_my_tool(
         self.require_parameter_descriptions = require_parameter_descriptions
         self.strict = strict
 
+    @classmethod
+    def from_schema(
+        cls,
+        function: Callable[..., Any],
+        name: str,
+        description: str,
+        json_schema: JsonSchemaValue,
+    ) -> Self:
+        """Creates a Pydantic tool from a function and a JSON schema.
+
+        Args:
+            function: The function to call.
+                This will be called with keywords only, and no validation of
+                the arguments will be performed.
+            name: The unique name of the tool that clearly communicates its purpose
+            description: Used to tell the model how/when/why to use the tool.
+                You can provide few-shot examples as a part of the description.
+            json_schema: The schema for the function arguments
+
+        Returns:
+            A Pydantic tool that calls the function
+        """
+        function_schema = _function_schema.FunctionSchema(
+            function=function,
+            description=description,
+            validator=SchemaValidator(schema=core_schema.any_schema()),
+            json_schema=json_schema,
+            takes_ctx=False,
+            is_async=asyncio.iscoroutinefunction(function),
+        )
+
+        return cls(
+            function,
+            takes_ctx=False,
+            name=name,
+            description=description,
+            function_schema=function_schema,
+        )
+
+    @classmethod
+    def from_langchain(cls, langchain_tool: LangChainTool) -> Self:
```

Comment:
What do you think about moving this off of `Tool` and into a new module like `pydantic_ai.ext.langchain`? We will likely have more wrappers and convenience methods for other frameworks in the future, so my colleague @Viicos correctly pointed out that we should probably start moving them into their own namespace already instead of cluttering up the regular public classes.

This wouldn't need to be class method in that case, I imagine:

```py
from pydantic_ai import Agent
from pydantic_ai.ext.langchain import langchain_tool
from langchain_community.tools import DuckDuckGoSearchRun

agent = Agent(..., tools=[langchain_tool(DuckDuckGoSearchRun])
```